metadata-ingestion/src/datahub/ingestion/source/kafka.py

```diff
@@ -265,10 +294,96 @@ def _extract_record(self, topic: str) -> Iterable[MetadataWorkUnit]:
                 self.report.report_workunit(wu)
                 yield wu
 
+    def build_custom_properties(self, topic_detail, extra_topic_config):
+
+        replication_factor = None
+        for _, p_meta in topic_detail.partitions.items():
+            if replication_factor is None or len(p_meta.replicas) > replication_factor:
+                replication_factor = len(p_meta.replicas)
+
+        MIN_INSYNC_REPLICAS_CONFIG = "min.insync.replicas"
+        RETENTION_SIZE_CONFIG = "retention.bytes"
+        RETENTION_TIME_CONFIG = "retention.ms"
+        CLEANUP_POLICY_CONFIG = "cleanup.policy"
+        MAX_MESSAGE_SIZE_CONFIG = "max.message.bytes"
+        UNCLEAN_LEADER_ELECTION_CONFIG = "unclean.leader.election.enable"
+
+        custom_props = {
+            "Partitions": len(topic_detail.partitions),
+            "Replication Factor": replication_factor,
+        }
+        if extra_topic_config is not None:
+            custom_props.update(
+                {
+                    "Retention Size (In Bytes)": self._get_config_value_if_present(
+                        extra_topic_config, RETENTION_SIZE_CONFIG
+                    ),
+                    "Retention Time (In Milliseconds)": self._get_config_value_if_present(
+                        extra_topic_config, RETENTION_TIME_CONFIG
+                    ),
+                    "Max Message Size (In Bytes)": self._get_config_value_if_present(
+                        extra_topic_config, MAX_MESSAGE_SIZE_CONFIG
+                    ),
+                    "Cleanup Policies": self._get_config_value_if_present(
+                        extra_topic_config, CLEANUP_POLICY_CONFIG
+                    ),
+                    "Minimum Number of In-Sync Replicas Required": self._get_config_value_if_present(
+                        extra_topic_config, MIN_INSYNC_REPLICAS_CONFIG
+                    ),
+                    "Unclean Leader Election Enabled": self._get_config_value_if_present(
+                        extra_topic_config, UNCLEAN_LEADER_ELECTION_CONFIG
+                    ),
+                }
+            )
+
+        return {
+            k: (json.dumps(v) if not isinstance(v, str) else v)
+            for k, v in custom_props.items()
+            if v is not None
+        }
+
     def get_report(self) -> KafkaSourceReport:
         return self.report
 
     def close(self) -> None:
         if self.consumer:
             self.consumer.close()
         super().close()
+
+    def _get_config_value_if_present(
+        self, config_dict: Dict[str, confluent_kafka.admin.ConfigEntry], key: str
+    ) -> Any:
+        return config_dict[key].value if key in config_dict.keys() else None
+
+    def fetch_extra_topic_details(self, topics: List[str]) -> Dict[str, dict]:
+        logger.debug("Fetching config details for all topics")
```

Comment:
I think this can be an info log to make sure the user sees what is happening

Reply:
Done